whisper.cpp

```diff
@@ -4082,6 +4096,79 @@ static void whisper_sequence_score(
     }
 }
 
+static bool whisper_fast_swap_KV(std::vector<int> & view, whisper_decoder src[], int size) {
+    // (decoder->buffer->decoder or decoder->buffer + decoder->decoder)
+    std::unordered_set<int> two_copy; // decoder indices require two copies to safely modify KV caches
+    two_copy.reserve(size);
+
+    // (buffer->decoder or decoder->decoder)
+    std::unordered_set<int> one_copy; // decoder indices require one copy to safely modify KV caches
+    one_copy.reserve(size);
+
+    for (int i = 0; i < size; i++) {
+        // zero-copy (no modification)
+        if (i == view[i] || view[i] < 0) {continue;}
+
+        bool is_one_copy = true;
+        // since we modify data sequentially, we only consider decoder indices after current index
+        for (int j = i + 1; j < size; j++) {
+            if (i == view[j]) {
+                two_copy.insert(i);
+                is_one_copy = false;
+                break;
+            }
+        }
+        if (is_one_copy) {
+            one_copy.insert(i);
+        }
+    }
+
+    // beam-search helpers
+    struct kv_buf {
+        std::vector<uint8_t> k;
+        std::vector<uint8_t> v;
+    };
+
+    std::vector<kv_buf> kv_bufs;
+    kv_bufs.resize(size);
+
+    for (auto &i : two_copy) {
```

Comment:
Respect the coding style:

```suggestion
    for (auto & i : two_copy) {
```

Fix in other places too